entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md

```diff
@@ -2064,3 +2064,558 @@ Configuration of stored procedures many-to-many join entities can be performed a
+
+EF7 makes it easier to handle dynamic connections and connection strings through improvements to the <xref:Microsoft.EntityFrameworkCore.Diagnostics.IDbConnectionInterceptor>. This starts with the ability to configure the `DbContext` without any connection string. For example:
+
+```csharp
+services.AddDbContext<CustomerContext>(
+    b => b.UseSqlServer();
+```
+
+One of the `IDbConnectionInterceptor` methods can then be implemented to configure the connection before it is used. `ConnectionOpeningAsync` is a good choice, since it can perform an async operation to obtain the connection string, find an access token, and so on. For example, imagine a service scoped to the current request that understands the current tenant:
+
+```csharp
+services.AddScoped<ITenantConnectionStringFactory, TestTenantConnectionStringFactory>();
+```
+
+This can be injected into each `DbContext` instance using constructor injection:
+
+```csharp
+public class CustomerContext : DbContext
+{
+    private readonly ITenantConnectionStringFactory _connectionStringFactory;
+
+    public CustomerContext(
+        DbContextOptions<CustomerContext> options,
+        ITenantConnectionStringFactory connectionStringFactory)
+        : base(options)
+    {
+        _connectionStringFactory = connectionStringFactory;
+    }
+
+    // ...
+}
+```
+
+This service is then used when constructing the interceptor implementation for the context:
+
+```csharp
+protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+    => optionsBuilder.AddInterceptors(
+        new ConnectionStringInitializationInterceptor(_connectionStringFactory));
+```
+
+Finally, the interceptor uses this service to obtain the connection string asynchronously and set it the first time that the connection is used:
+
+<!--
+    public class ConnectionStringInitializationInterceptor : DbConnectionInterceptor
+    {
+        private readonly ITenantConnectionStringFactory _connectionStringFactory;
+
+        public ConnectionStringInitializationInterceptor(ITenantConnectionStringFactory connectionStringFactory)
+        {
+            _connectionStringFactory = connectionStringFactory;
+        }
+
+        public override InterceptionResult ConnectionOpening(
+            DbConnection connection,
+            ConnectionEventData eventData,
+            InterceptionResult result)
+            => throw new NotSupportedException("Synchronous connections not supported.");
+
+        public override async ValueTask<InterceptionResult> ConnectionOpeningAsync(
+            DbConnection connection, ConnectionEventData eventData, InterceptionResult result,
+            CancellationToken cancellationToken = new())
+        {
+            if (string.IsNullOrEmpty(connection.ConnectionString))
+            {
+                connection.ConnectionString = (await _connectionStringFactory.GetConnectionStringAsync(cancellationToken));
```

Comment:
It's really hard for me to imagine a situation where someone should do this...

Reply:
See previous comment.